core/templates/filters/string-utility-filters/truncate-and-capitalize.pipe.ts

```diff
@@ -21,11 +21,16 @@ import { Pipe, PipeTransform } from '@angular/core';
 // Note that this filter does not truncate at the middle of a word.
 @Pipe({name: 'truncateAndCapitalize'})
 export class TruncateAndCapitalizePipe implements PipeTransform {
-  transform(input: string, maxNumberOfCharacters: number): string {
+  // 'maxNumberOfCharacters' can be set to 'null' for returning the entire input
+  // string with the first letter capitalized.
+  transform(input: string, maxNumberOfCharacters: number | null): string {
```

Comment:
Is there a use case in codebase where `maxNumberOfCharacters` has been passed as null ?(Apart from the tests)

Reply:
I could not find any usecase of this pipe inside the codebase, but having a None value for `maxNumberOfCharacters` makes sense. We can also refactor it and make `maxNumberOfCharacters` an optional parameter but a `CapitalizePipe` already exists for this purpose.

Reply:
This is used in `collection-summary-tile` , `exploration-summary-tile` and `information-card-modal` but with `maxNumberOfCharacters` as a parameter. I also think that this property can be null as mentioned in the test but making this optional would be better than making it null? I'm not sure if we make this ' | null ' then we have to use it like {{ .. | truncateAndCapitalize: null }} or just {{ .. | truncateAndCapitalize }} , can you confirm?

Reply:
I think making it optional brings a certain level of ambiguity with what the pipe is doing. When making it `| null`, we would have to specify an argument for the parameter since `null` is a value. So, we will have to use it like ` {{ .. | truncateAndCapitalize: null }} `. 

In any case, keeping the `null` value makes sense logically. wdyt?

Reply:
One would prefer to use `CapitalizePipe` instead of this notation `{{ .. | truncateAndCapitalize: null }}` then , I think the respective test should be removed here and we should make this property mandatory? @vojtechjelinek wdyt about this?

Reply:
I still think the particular test case should be removed, I'm approving to unblock my review!!

Reply:
Yeah, lets make it required.